platform/src/pages/account/login/index.jsx

```diff
@@ -13,18 +17,41 @@ const UserLogin = () => {
 
   const handleLogin = async (e) => {
     e.preventDefault();
-    try {
-      await dispatch(loginUser(postData.userData)).then(()=>{
-        if (postData.success) {
-          router.push('/analytics');
-        } else {
-          setErrors(true);
-          setError(postData.errors);
-        }
+    try{
+      await postUserLoginDetails(postData.userData)
+      .then((res) => {
+          const { token } = res;
+          localStorage.setItem('token', token);
+          setAuthToken(token);
+          // Decode token to get user data
+          const decoded = jwt_decode(token);
+          localStorage.setItem('loggedUser', JSON.stringify(decoded));
+          dispatch(setUserInfo(decoded))
+          dispatch(setSuccess(true))
+          router.push('/analytics')
       })
-    } catch (err) {}
+      .catch((error) => {
+          dispatch(setSuccess(false))
```

Comment:
thanks for this @FaithDaka .....this looks good and reminds me of our most recent quick sync.

Could we do this for the `Gmail Login` functionality also? And handle the `manual redirection` (302 http error code) from the .catch() block of this Axios Promise chain.

**WHY?**

1. This handles the errors specific to the API request itself or any other issues that occur during this asynchronous operations.
2. We still maintain the try{} catch() block outside the entire API request call to handle the synchronous code within the function to handle any errors which happen here.

I am not necessarily resolve our CORS issues but it definitely sounds like a good approach to take --- that is, utilising a combination of these error handling mechanisms to enable us handle both the asynchronous and the synchronous errors in our code.

Thanks!

Reply:
@Baalmart For the _google-auth_, the actual error is a CORS error. 
The **302** is the status code (you keep calling it an error 😅 ) returned in the preflight response which is telling the frontend to redirect. So when it tries to redirect, a CORS error is thrown back and that is what is caught in the _catch{}_ statement.